src/cpu/aarch64/acl_post_ops.hpp

```diff
@@ -0,0 +1,226 @@
+/*******************************************************************************
+* Copyright 2022 Arm Ltd. and affiliates
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_AARCH64_ACL_POST_OPS_HPP
+#define CPU_AARCH64_ACL_POST_OPS_HPP
+
+#include "cpu/cpu_convolution_pd.hpp"
+
+#include "cpu/aarch64/acl_binary.hpp"
+#include "cpu/aarch64/acl_eltwise.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+struct acl_post_ops_conf_t {
+    // Index of the sum post op if there is one, < 0 means no sum
+    int sum_index = -1;
+};
+
+struct acl_post_ops_t : public primitive_t {
+    struct pd_t : public primitive_desc_t {
+        // Construct the primitive desc for an acl_post_ops_t from a post_ops_t
+        // and the memory desc for the destination. Note that if a post op is
+        // fused into the base primitive, then it will need to be removed from the
+        // post ops before passing in. This can be done after construction using
+        // the zero argument constructor.
+        pd_t(const post_ops_t &post_ops)
+            : primitive_desc_t(primitive_kind::undefined) {
```

Comment:
`undefined` is kind of a way to find an error if `primitive_kind` is not initialized.
Please use `zero_pad` as an example of an internal primtive kind. This will allow to keep architecture consistent. For example, this will enable primitive/primitive_desc caches in case the same `acl_post_ops_t` is used multiple times.
New primitive_kind can be called `post_ops` or something like that.
For internal only primitive kinds a new `primitive_kind` can be defined here: https://github.com/oneapi-src/oneDNN/blob/bbba4757a45e96be5ab0b9781c155f6447c1e708/src/common/c_types_map.hpp#L1265 
There are some other places where it should be added (there are more):
https://github.com/oneapi-src/oneDNN/blob/bbba4757a45e96be5ab0b9781c155f6447c1e708/src/common/c_types_map.hpp#L1395
https://github.com/oneapi-src/oneDNN/blob/bbba4757a45e96be5ab0b9781c155f6447c1e708/src/common/c_types_map.hpp#L1367
https://github.com/oneapi-src/oneDNN/blob/bbba4757a45e96be5ab0b9781c155f6447c1e708/src/common/c_types_map.hpp#L1426
https://github.com/oneapi-src/oneDNN/blob/bbba4757a45e96be5ab0b9781c155f6447c1e708/src/common/type_helpers.hpp#L903

Reply:
Thank you for the review. I have added a new internal only `primitive_kind_t` and (I think!) all the necessary related machinery.